x-pack/plugins/lens/public/datatable_visualization/expression.tsx

```diff
@@ -214,6 +264,120 @@ export function DatatableComponent(props: DatatableRenderProps) {
+                            value: `${fieldName ? `${fieldName}: ` : ''}${formattedValue}`,
+                          },
+                        })}
+                        data-test-subj="lensDatatableFilterFor"
+                        onClick={() => handleFilterClick(field, value, colIndex)}
+                      />
+                    </EuiToolTip>
+                    <EuiFlexItem grow={false}>
+                      <EuiToolTip
+                        position="bottom"
+                        content={i18n.translate('xpack.lens.excludeValueButtonTooltip', {
+                          defaultMessage: 'Exclude value',
+                        })}
+                      >
+                        <EuiButtonIcon
+                          iconType="minusInCircle"
+                          color="text"
+                          aria-label={i18n.translate('xpack.lens.excludeValueButtonAriaLabel', {
+                            defaultMessage: `Exclude {value}`,
+                            values: {
+                              value: `${fieldName ? `${fieldName}: ` : ''}${formattedValue}`,
+                            },
+                          })}
+                          data-test-subj="lensDatatableFilterOut"
+                          onClick={() => handleFilterClick(field, value, colIndex, true)}
+                        />
+                      </EuiToolTip>
+                    </EuiFlexItem>
+                  </EuiFlexGroup>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            );
+          }
+          return <span data-test-subj="lnsDataTableCellValue">{formattedValue}</span>;
+        },
+      };
+    })
+    .filter(({ field }) => !!field);
+
+  if (!!props.rowHasRowClickTriggerActions && !!onRowContextMenuClick) {
+    const hasAtLeastOneRowClickAction = props.rowHasRowClickTriggerActions.find((x) => x);
+    if (hasAtLeastOneRowClickAction) {
+      const actions: EuiTableActionsColumnType<unknown> = {
+        name: 'Actions',
+        actions: [
+          {
+            name: i18n.translate('xpack.lens.tableRowMore', {
+              defaultMessage: 'More',
+            }),
+            description: i18n.translate('xpack.lens.tableRowMoreDescription', {
+              defaultMessage: 'Table row context menu',
+            }),
+            type: 'icon',
+            icon: ({ rowIndex }: { rowIndex: number }) => {
+              if (
+                !!props.rowHasRowClickTriggerActions &&
+                !props.rowHasRowClickTriggerActions[rowIndex]
+              )
+                return 'empty';
+              return 'boxesVertical';
+            },
+            onClick: ({ rowIndex }) => {
+              onRowContextMenuClick({
+                rowIndex,
+                table: firstTable,
+                columns: props.args.columns.columnIds,
+              });
+            },
+          },
+        ],
+      } as EuiTableActionsColumnType<unknown>;
```

Comment:
Can we improve the typing here? It seems like we can turn `EuiBasicTableColumn<unknown>` into `EuiBasicTableColumn<{ rowIndex: number; [key: string]: unknown }>`. This would also remove the need to explicitly cast here

Reply:
Done.